### DISCUSSION TITLE
Pass data/state to the embedded app

### DISCUSSION THREAD

**Discussion by tomas7** [vendurehq/vendure#1280] [General]
It would be very helpful to have an option to pass data (either product slug or id) to addActionBarItem as a payload or parameter and access it in the embedded app (React). 

My user case: Venture admin-UI is extended by react-app. React component is fetching data from another API based on selected product (slug/id), button addActionBarItem is located at product-details. When I try to pass the ID to the URL, react component can't access it. 

![image](https://user-images.githubusercontent.com/11965514/145566131-21429f9c-c036-4238-bea1-e0933771c135.png)

Is there any workaround?

Request created based on @michaelbromley from Vendure Slack Channel.

Original comment: _We are lacking an API that allows you to easily access state from your embedded app.
This can be added though - we can use the same PostMessage channels that we already use to securely pass data to the data service & notification service. Could you open a feature request for this?_

Thank you for any help,
Tomas

**Comment by michaelbromley** [comment]
Issue resolved in https://github.com/vendure-ecommerce/vendure/issues/1281
